compiler-core/src/type_.rs

```diff
@@ -134,6 +134,28 @@ impl Type {
         }
     }
 
+    pub fn tuple_types(&self) -> Option<Vec<Arc<Self>>> {
+        match self {
+            Self::Tuple { elems } => Some(elems.clone()),
+            _ => None,
+        }
+    }
+
+    pub fn constructor_types(&self) -> Option<Vec<Arc<Self>>> {
+        match self {
+            Self::Named { args, .. } => Some(args.clone()),
+            Self::Var { type_, .. } => type_.borrow().constructor_types(),
+            _ => None,
+        }
+    }
+
+    pub fn nested_type(&self) -> Option<Arc<Self>> {
```

Comment:
What's a nested type?

Could you document these three functions please 🙏

Reply:
I'm realizing inner_type is probably a better name for this. I just needed a way to follow the Var to the underlying type so we had proper linking. Lemme rename and also add comments